qiskit/primitives/utils.py

```diff
@@ -143,6 +143,17 @@ def _circuit_key(circuit: QuantumCircuit, functional: bool = True) -> tuple:
     )
 
 
+def _observable_key(observable: SparsePauliOp) -> tuple:
```

Comment:
CI error indicates that you need to add a path for `PauliSumOp`.
```
    AttributeError: 'PauliSumOp' object has no attribute 'to_list'
```

Reply:
The mistake was not converting to SparsePauliOp at this stage. I have fixed.